### PR TITLE
Miscellaneous inference bug fixes

### PR DESCRIPTION
**Pull request by santhnm2**
# What does this PR do ?

Makes the following bug fixes:
1. Decrements the `active_request_count` and `total_request_count` for the chunked prefill request in `update_requests` rather than manually adjusting the `active_token_count` in `add_request`. This requires swapping the chunked prefill request even if it is currently inactive.
2. Ensure that the partial chunked prefill request data is cleared in the event of KV cache recompute. Also, on resume ensure the chunked prefill request is the first request scheduled.
3. Handle additional edge case that could lead to prompt length of 1 with chunked prefill. 
4. Corrects the `seq_idx` construction in `MambaMetadata` in the event of `paused_request_count > 0`.
5. Sets the Mamba state tensor dtypes to fp32 for the chunked prefill functional test to ensure robustness.

### Pre-checks

- [ ] I have added relevant unit tests
- [ ] I have added relevant functional tests
- [ ] I have added proper typing to my code [Typing guidelines](https://docs.python.org/3/library/typing.html)
- [ ] I have added relevant documentation
- [ ] I have run the [autoformatter.sh](https://github.com/NVIDIA/Megatron-LM/blob/main/tools/autoformat.sh) on my PR

### Code review

Feel free to message or comment the [@mcore-oncall](https://github.com/orgs/NVIDIA/teams/mcore-oncall) to help accelerate your merge into main. The less complex your PR is, the faster it will be approved and merged!

All PRs start as **draft**. If you open a non-draft PR, it will be automatically converted to draft.

#### Step 1: Mark PR as "Ready for Review"

1. When your PR is ready, click **Ready for Review**.
2. An oncall reviewer is auto-assigned and expert reviewers are notified based on your changes.
   - Some PRs may jump straight to step 2. This is determined by `.github/CODEOWNERS`.

:warning: Only mark as ready once merge-conflicts are resolved and the CI is passing.
Final Review might get declined if these requirements are not fulfilled.

#### Step 2: Final Review

For PRs that change `megatron/core`, once all expert reviewers have approved, the `Final Review` label is applied **automatically** and final reviewers are assigned.

For PRs outside `megatron/core`, this step is skipped.

#### Step 3: Approved

Once all required reviewers have approved, the `Approved` label is applied **automatically**.

### Merge

Any member of [mcore-engineers](https://github.com/orgs/NVIDIA/teams/mcore-engineers) will be able to merge your PR.

<details>
<summary>For MRs into `dev` branch</summary>
The proposed review process for `dev` branch is under active discussion.

MRs are mergable after one approval by either `eharper@nvidia.com` or `zijiey@nvidia.com`.
</details>

### CODE REVIEW

File: megatron/core/inference/contexts/dynamic_context.py [hunk 1]
```diff
@@ -2556,14 +2552,20 @@ def update_requests(
...
+                src_idxs=torch.tensor([chunked_prefill_request_idx]),
                 dst_idxs=torch.tensor([self.total_request_count - 1]),
                 next_tokens=next_tokens,
                 new_speculative_tokens=new_speculative_tokens,
             )
 
+            # Explicitly decrement the active and total request counts here so that the chunked
+            # prefill request metadata is not updated. This will all be restored when the next
+            # chunk is added through add_request.
+            active_request_count -= 1
```
Comment (kvareddy) [inline megatron/core/inference/contexts/dynamic_context.py]: This should be conditional on where chunked prefill request is present or not.

**Review by kvareddy — APPROVED** [review]
